src/systems/pen-tools.js

```diff
@@ -0,0 +1,43 @@
+import { waitForDOMContentLoaded } from "../utils/async-utils";
+
+// Used for tracking and managing pen tools in the scene
+AFRAME.registerSystem("pen-tools", {
```

Comment:
You may also need to listen to alternative event for knowing when to update the place UI: 
https://github.com/mozilla/hubs/blob/redesign/toolbar-buttons/src/react-components/room/PlacePopoverContainer.js#L100

Reply:
I like the current implementation because it allows for removing a pen that you own but have dropped. Also this code is a direct copy of the `camera-tools` system.